Mathlib/AlgebraicTopology/SimplicialSet/Coskeletal.lean

```diff
@@ -0,0 +1,399 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl, Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.StrictSegal
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+
+/-!
+# Coskeletal simplicial sets
+
+The identity natural transformation exhibits a simplicial set `X` as a right extension of its
+restriction along `(Truncated.inclusion (n := n)).op` recorded by `rightExtensionInclusion X n`.
+
+The simplicial set `X` is *n-coskeletal* if
+`(rightExtensionInclusion X n).IsPointwiseRightKanExtension` holds.
+
+In this file, we prove that if `X` is `StrictSegal` then `X` is 2-coskeletal defining
+
+`StrictSegal.cosk2Iso : X ≅ (Truncated.cosk 2).obj ((truncation 2).obj X)`
+
+In particular, nerves of categories are 2-coskeletal.
+-/
+
+universe v u
+
+open CategoryTheory Simplicial SimplexCategory Opposite Category Functor Limits
+
+namespace SSet
+
+/-- The identity natural transformation exhibits a simplicial set as a right extension of its
+restriction along `(Truncated.inclusion (n := n)).op`.-/
+@[simps!]
+def rightExtensionInclusion (X : SSet.{u}) (n : ℕ) :
+    RightExtension (Truncated.inclusion (n := n)).op
+      (Functor.op Truncated.inclusion ⋙ X) := RightExtension.mk _ (𝟙 _)
+
+section
+
+local notation (priority := high) "[" n "]" => SimplexCategory.mk n
+
+set_option quotPrecheck false
+local macro:max (priority := high) "[" n:term "]₂" : term =>
+  `((⟨SimplexCategory.mk $n, by decide⟩ : SimplexCategory.Truncated 2))
+
+/-- The object of `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` corresponding to the
+map [0] ⟶ [n] with image `i`. -/
+private
+def pt {n} (i : Fin (n + 1)) : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op :=
```

Comment:
The full name of this definition is `SSet.pt`. That sounds a bit too generic to me...
Should it maybe be in the `StructuredArrow` namespace instead?

Reply:
Noted, but now moved to a different PR (which I'll open after this is merged).